core/server/master/src/main/java/alluxio/master/AlluxioMasterProcess.java

```diff
@@ -305,6 +305,16 @@ protected void startServingRPCServer() {
           GrpcServerAddress.create(mRpcConnectAddress.getHostName(), mRpcBindAddress),
           ServerConfiguration.global(), ServerUserState.global());
 
+      int parallelism = ServerConfiguration.getInt(PropertyKey.MASTER_RPC_EXECUTOR_PARALLELISM);
```

Comment:
This is a lot of checking of arguments. Should this go in the constructor, or earlier? This is in the serving method, which secondaries will not even run until much later.

Reply:
Yea I moved this check up to the `Factory.create()`.